src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -170,6 +184,13 @@ public class SqlMutationEngine : IMutationEngine
                 throw _dabExceptionWithTransactionErrorMessage;
             }
 
+            if (!isReadPermissionConfigured)
+            {
+                throw new DataApiBuilderException(message: "The database operation was successful but the current user is unable to view the response due to lack of read permissions",
```

Comment:
```suggestion
                throw new DataApiBuilderException(message: "The mutation operation was successful but the current user is unable to view the response due to lack of read permissions",
```

Reply:
beat me to this suggestion!

Reply:
Perhaps the end of the message should read:
> but the current user is unauthorized to view the response.

Reply:
I think it is worth explicitly calling out the lack of `read` permissions - as it is easy to think that just create permission should be sufficient for create mutation while actually we need create + read permissions.

So, I've updated the error message to - `The mutation operation was successful but the current user is unauthorized to view the response due to lack of read permissions`